Updated weather agent for WSU. Updated with review comments

schandrika opened the pull request:
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/VOLTTRON/volttron/pull/1727%23issuecomment-402332606%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1727%23issuecomment-402332644%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1727%23issuecomment-402525707%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1727%23discussion_r201101795%22%2C%20%22https%3A//github.com/VOLTTRON/volttron/pull/1727%23discussion_r201159022%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1727%23issuecomment-402332606%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20get%20the%20following%20error%20when%20running%20an%20installed%20weather%20agent.%5Cr%5Cn%5Cr%5Cn%60%60%60%60%20bash%5Cr%5CnTraceback%20%28most%20recent%20call%20last%29%3A%5Cr%5Cn%20%20File%20%5C%22/home/osboxes/git/volttron/volttron/platform/vip/agent/core.py%5C%22%2C%20line%20109%2C%20in%20_loop%5Cr%5Cn%20%20%20%20method%28%2Aself.args%2C%20%2A%2Aself.kwargs%29%5Cr%5Cn%20%20File%20%5C%22/home/osboxes/.volttron/agents/1679c664-d3c8-41c9-a366-cc59a75a48a9/weatheragent-2.0/weather/weather.py%5C%22%2C%20line%20348%2C%20in%20on_polling%5Cr%5Cn%20%20%20%20for%20location%20in%20self.locations%3A%5Cr%5CnTypeError%3A%20%27NoneType%27%20object%20is%20not%20iterable%5Cr%5Cn%60%60%60%60%5Cr%5Cn%5Cr%5CnI%20used%20the%20following%20config%3A%5Cr%5Cn%5Cr%5Cn%60%60%60%60%20json%5Cr%5Cn%5Cr%5Cn%7B%5Cr%5Cn%20%20%20%20%5C%22operation_mode%5C%22%3A%202%2C%20%231%20request%20only%20%2C%202%20poll%20and%20requests%5Cr%5Cn%5Cr%5Cn%20%20%20%20%5C%22wu_api_key%5C%22%3A%20%5C%22%3Capi_key%3E%5C%22%2C%5Cr%5Cn%5Cr%5Cn%20%20%20%20%5C%22location%5C%22%3A%20%5B%5Cr%5Cn%20%20%20%20%20%20%20%20%7B%5Cr%5Cn%20%20%20%20%20%20%20%20%20%20%20%20%5C%22zip%5C%22%3A%20%5C%2299336%5C%22%2C%5Cr%5Cn%20%20%20%20%20%20%20%20%7D%2C%5Cr%5Cn%20%20%20%20%20%20%20%20%7B%5Cr%5Cn%20%20%20%20%20%20%20%20%20%20%20%20%5C%22zip%5C%22%3A%20%5C%2299354%5C%22%5Cr%5Cn%20%20%20%20%20%20%20%20%7D%5Cr%5Cn%20%20%20%20%5D%2C%5Cr%5Cn%20%20%20%20%5C%22poll_time%5C%22%3A%2020%20%23seconds%5Cr%5Cn%5Cr%5Cn%7D%5Cr%5Cn%60%60%60%60%5Cr%5Cn%5Cr%5CnNote%20please%20reformat%20the%20default%20config%20and%20the%20embeded%20config%20in%20the%20rst%20read%20me%20file.%22%2C%20%22created_at%22%3A%20%222018-07-04T01%3A01%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/3979063%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/craig8%22%7D%7D%2C%20%7B%22body%22%3A%20%22Note%20I%20did%20use%20a%20valid%20api%20key%22%2C%20%22created_at%22%3A%20%222018-07-04T01%3A02%3A10Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/3979063%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/craig8%22%7D%7D%2C%20%7B%22body%22%3A%20%22Sorry.%20typo.%20The%20configuration%20must%20have%20locations%20instead%20of%20location.%20I%20fixed%20this%2C%20updated%20README%20and%20tested%20by%20installing%20agent%20with%20two%20zip%20code%20in%20configuration%20file.%20%22%2C%20%22created_at%22%3A%20%222018-07-04T16%3A50%3A57Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20f16c414feaa40dc3f1df14d1fb409830760c7d02%20volttron/platform/messaging/topics.py%2010%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1727%23discussion_r201159022%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22this%20was%20already%20there%20in%20topic.py%22%2C%20%22created_at%22%3A%20%222018-07-09T21%3A52%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars2.githubusercontent.com/u/16124920%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/schandrika%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20volttron/platform/messaging/topics.py%3AL191-202%22%7D%2C%20%22Pull%20f16c414feaa40dc3f1df14d1fb409830760c7d02%20services/core/WeatherAgent/setup.py%206%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/VOLTTRON/volttron/pull/1727%23discussion_r201101795%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20is%20the%20copyright%20older%20than%20it%20should%20be.%20%20Please%20upgrade%20all%20copyrights%20in%20this%20agent.%22%2C%20%22created_at%22%3A%20%222018-07-09T18%3A24%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars3.githubusercontent.com/u/3979063%3Fv%3D4%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/craig8%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20services/core/WeatherAgent/setup.py%3AL1-66%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'><img src='http://www.codereviewhub.com/site/github-bar.png' height=40></a>
- [ ] <a href='#crh-comment-General Comment'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1727#issuecomment-402332606'>General Comment</a></b>
- <a href='https://github.com/craig8'><img border=0 src='https://avatars3.githubusercontent.com/u/3979063?v=4' height=16 width=16></a> I get the following error when running an installed weather agent.
```` bash
Traceback (most recent call last):
File "/home/osboxes/git/volttron/volttron/platform/vip/agent/core.py", line 109, in _loop
method(*self.args, **self.kwargs)
File "/home/osboxes/.volttron/agents/1679c664-d3c8-41c9-a366-cc59a75a48a9/weatheragent-2.0/weather/weather.py", line 348, in on_polling
for location in self.locations:
TypeError: 'NoneType' object is not iterable
````
I used the following config:
```` json
{
"operation_mode": 2, #1 request only , 2 poll and requests
"wu_api_key": "<api_key>",
"location": [
{
"zip": "99336",
},
{
"zip": "99354"
}
],
"poll_time": 20 #seconds
}
````
Note please reformat the default config and the embeded config in the rst read me file.
- <a href='https://github.com/craig8'><img border=0 src='https://avatars3.githubusercontent.com/u/3979063?v=4' height=16 width=16></a> Note I did use a valid api key
- <a href='https://github.com/schandrika'><img border=0 src='https://avatars2.githubusercontent.com/u/16124920?v=4' height=16 width=16></a> Sorry. typo. The configuration must have locations instead of location. I fixed this, updated README and tested by installing agent with two zip code in configuration file.
- [ ] <a href='#crh-comment-Pull f16c414feaa40dc3f1df14d1fb409830760c7d02 services/core/WeatherAgent/setup.py 6'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1727#discussion_r201101795'>File: services/core/WeatherAgent/setup.py:L1-66</a></b>
- <a href='https://github.com/craig8'><img border=0 src='https://avatars3.githubusercontent.com/u/3979063?v=4' height=16 width=16></a> Why is the copyright older than it should be.  Please upgrade all copyrights in this agent.
- [ ] <a href='#crh-comment-Pull f16c414feaa40dc3f1df14d1fb409830760c7d02 volttron/platform/messaging/topics.py 10'></a> <img src='http://www.codereviewhub.com/site/github-remaining.png' height=16 width=60>&nbsp;<b><a href='https://github.com/VOLTTRON/volttron/pull/1727#discussion_r201159022'>File: volttron/platform/messaging/topics.py:L191-202</a></b>
- <a href='https://github.com/schandrika'><img border=0 src='https://avatars2.githubusercontent.com/u/16124920?v=4' height=16 width=16></a> this was already there in topic.py


<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1727?mark_as_completed=1'><img src='http://www.codereviewhub.com/site/github-mark-as-completed.png' height=26></a>&nbsp;<a href='https://www.codereviewhub.com/VOLTTRON/volttron/pull/1727?approve=1'><img src='http://www.codereviewhub.com/site/github-approve.png' height=26></a>&nbsp;<a href='https://github.com/VOLTTRON/volttron/pull/1727'><img src='http://www.codereviewhub.com/site/github-refresh.png' height=26></a>
<a href='#crh-end'></a>

craig8 commented:
This pull request is now managed with CodeReviewHub. <a href='https://github.com/VOLTTRON/volttron/pull/1727'>Refresh</a>.<a href='#crh-update-none'></a>

craig8 commented:
I get the following error when running an installed weather agent.

```` bash
Traceback (most recent call last):
  File "/home/osboxes/git/volttron/volttron/platform/vip/agent/core.py", line 109, in _loop
    method(*self.args, **self.kwargs)
  File "/home/osboxes/.volttron/agents/1679c664-d3c8-41c9-a366-cc59a75a48a9/weatheragent-2.0/weather/weather.py", line 348, in on_polling
    for location in self.locations:
TypeError: 'NoneType' object is not iterable
````

I used the following config:

```` json

{
    "operation_mode": 2, #1 request only , 2 poll and requests

    "wu_api_key": "<api_key>",

    "location": [
        {
            "zip": "99336",
        },
        {
            "zip": "99354"
        }
    ],
    "poll_time": 20 #seconds

}
````

Note please reformat the default config and the embeded config in the rst read me file.

craig8 commented:
Note I did use a valid api key

schandrika commented:
Sorry. typo. The configuration must have locations instead of location. I fixed this, updated README and tested by installing agent with two zip code in configuration file.

schandrika commented on volttron/platform/messaging/topics.py:
```diff
@@ -191,5 +191,11 @@
+WEATHER2_RESPONSE = 'weather2/response'
+WEATHER2_ERROR = 'weather2/error'
+
 BACNET_INFO_BASE = _('bacnet/{indication}')
```
this was already there in topic.py
